crates/sui-types/src/message_envelope.rs

```diff
@@ -124,7 +111,128 @@ where
             )?,
         };
 
-        cert.verify(committee)?;
         Ok(cert)
     }
 }
+
+/// TrustedEnvelope is a serializable wrapper around Envelope which is
+/// Into<VerifiedEnvelope> - in other words it models a verified message which has been
+/// written to the db (or some other trusted store), and may be read back from the db without
+/// further signature verification.
+///
+/// TrustedEnvelope should *only* appear in database interfaces.
+///
+/// DO NOT USE in networked APIs.
+///
+/// Because it is used very sparingly, it can be audited easily: Use rust-analyzer,
+/// or run: git grep -E 'TrustedEnvelope'
+///
+/// And verify that none of the uses appear in any network APIs.
+#[derive(Clone, Serialize, Deserialize)]
+pub struct TrustedEnvelope<T: Message, S>(Envelope<T, S>);
```

Comment:
Generally you should avoid the bounds here (unless you absolutely need to) so in this case you should do the following:
```suggestion
pub struct TrustedEnvelope<T, S>(Envelope<T, S>);
```

Reply:
This means you can drop the `T: Message` bounds on things like the Debug impl which aren't necessary

Reply:
So I had to add `Message` to `T` because `Envelope` is bound to `T`. And `Envelope` is bound to `T` because it has a field that uses dependent types that's bound to `Message`..

Reply:
ah...yes i had missed that...Although looking at this it appears as if `Envelope` can't be used for Transactions because Transactions include the user signatures in their digest?